### ISSUE TITLE
Generics - new() Type constraint doesn't work

### ISSUE THREAD

**Issue by calcock** [bridgedotnet/Bridge#652]
I'm having a problem with classes using Generics and the new() constraint, getting an error "Cannot find type: `0" at compilation time, caused by code creating an instance of the generic type.  A reproduce case can be found [here](http://live.bridge.net/#9bc5ef86837c06949a05) which is the same as the below.

```
public class App
{
    [Ready]
    public static void Main()
    {
       var x = new Wibble();
    }
}
public class Wibble : Foo<Something> {}
public class Something {}
public class Foo<T> where T: new()
{
    protected T Bar;

    public Foo() {
       // This Line Causes Error: // Line 0, Col 0 : Cannot find type: `0
       Bar = new T();
    }
}
```

**Comment by calcock** [comment]
I also tried adding code for a parameterless constructor to the 'Something' class  - this made no difference

**Comment by LeonidVeriga** [comment]
@calcock, thank you for the reporting. We will include the fix into `1.10.2`
